backend/routes/content.py

```
@@ -391,7 +437,98 @@ def mark_module_complete_and_open_next(module_id, user_id):
+        else:
+            logger.error(f"Module is not a challenge for module_id {module_id}")
+            return None, "Module is not a challenge", 400
+
+        if solution_module is None:
+            logger.error(f"Solution not found for module_id {module_id}")
+            return None, "Solution not found", 404
+
+        logger.debug(f"Solution module found: {solution_module.id}")
+        return solution_module, None, 200
+
+    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)